gson/src/main/java/com/google/gson/JsonElement.java

```diff
@@ -259,14 +259,17 @@ public byte getAsByte() {
   }
 
   /**
-   * convenience method to get this element as a primitive character value.
+   * convenience method to get the first character of this element as a string or the first
+   * character of this array's first element as a string.
    *
-   * @return get this element as a primitive char value.
+   * @return the first character of the string.
    * @throws ClassCastException if the element is of not a {@link JsonPrimitive} and is not a valid
-   * char value.
+   * string value.
    * @throws IllegalStateException if the element is of the type {@link JsonArray} but contains
    * more than a single element.
    * @since 1.3
+   * @deprecated This method is misleading, as it does not get this element as a char but rather as
+   * a string's first character.
    */
   public char getAsCharacter() {
```

Comment:
I'm still wary of only using the Javadoc tag. Can you include the annotation as well?

Reply:
yep.

Reply:
i can't add a description in the annotation, so i'll leave the javadoc tag, too, if that's alright.